stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -0,0 +1,404 @@
+                    device.min_volume,
+                    picking.volume,
+                    device.max_volume,
+                )
+                and tools.float_compare(
+                    device.max_weight,
+                    picking.weight,
+                    precision_digits=self._precision_volume(),
+                )
+                > 0
+            ):
+                return device
+        return None
+
+    def _volume_condition_for_device_choice(
+        self, min_volume, picking_volume, max_volume
+    ):
+        lower_bound = tools.float_compare(picking_volume, min_volume, 2) >= 0
+        upper_bound = tools.float_compare(max_volume, picking_volume, 2) >= 0
+        return lower_bound and upper_bound
+
+    def _raise_create_batch_not_possible(self):
+        """Raise the correct error to desrcibe why it's not possible to
+        create a batch."""
+        # At this stage we know that there is no picking to process
+        # at least for the list of devices. To provide to the user
+        # the right information about the reason why there is no picking
+        # we check if there is a candidate pickings without device
+        # constrains. If not, we raise an error to inform the user that there
+        # is no picking to process otherwise we raise an error to inform the
+        # user that there is not suitable device to process the pickings.
+        domain = OR(
+            [
+                self._get_picking_domain_common(),
+                self._get_picking_domain_common(self.user_id),
+            ]
+        )
+        limit = 1
+        if self.add_picking_list_in_error:
+            limit = None
+        candidates = self.env["stock.picking"].search(domain, limit=limit)
+        if candidates:
+            pickings = candidates if self.add_picking_list_in_error else None
+            raise NoSuitableDeviceError(pickings=pickings)
+        raise NoPickingCandidateError()
+
+    def _create_batch(self, raise_if_not_possible=False):
+        """Create a batch transfer."""
+        self._reset_counters()
+        # first we try to get the first picking for the user
+        first_picking = self._get_first_picking(user=self.user_id)
+        if not first_picking:
+            # if there is no picking for the user, we get the first picking
+            # without user
+            first_picking = self._get_first_picking()
+        if not first_picking:
+            if raise_if_not_possible:
+                self._raise_create_batch_not_possible()
+            return self.env["stock.picking.batch"].browse()
+        device = self._compute_device_to_use(first_picking)
+        self._init_counters(first_picking, device)
+        self._apply_limits()
+        vals = self._create_batch_values()
+        batch = self.env["stock.picking.batch"].create(vals)
+        # Propagate user on selected pickings
+        user = self.user_id or self.env.user
+        batch.picking_ids.write({"user_id": user.id})
```

Comment:
As there is an explicit user on the wizard, only this one should be used. If you don't set a user on the wizard, then the batch will contain only pickings without responsible/user and the batch should not be assigned to a user
```suggestion
        if self.user_id:
            batch.picking_ids.write({"user_id": user.id})
```